plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -4488,15 +4465,20 @@ public void testTypeCoercionOnCreateTable()
         testCharCoercionOnCreateTable("CHAR 'ABc'", "'ABc'");
     }
 
-    private void testTimestampCoercionOnCreateTable(@Language("SQL") String actualValue, @Language("SQL") String expectedValue)
+    @Test
+    public void testTimestampCoercionOnCreateTable()
     {
-        try (TestTable testTable = newTrinoTable(
-                "test_timestamp_coercion_on_create_table",
-                "(ts TIMESTAMP)")) {
-            assertUpdate("INSERT INTO " + testTable.getName() + " VALUES (" + actualValue + ")", 1);
-            assertThat(getColumnType(testTable.getName(), "ts")).isEqualTo("timestamp(6)");
-            assertQuery("SELECT * FROM " + testTable.getName(), "VALUES " + expectedValue);
-            assertTimestampNtzFeature(testTable.getName());
+        for (TypeCoercionTestSetup setup : typeCoercionTimestampProvider()) {
+            try (TestTable testTable = newTrinoTable(
+                    "test_timestamp_coercion_on_create_table",
+                    "(ts %s)".formatted(setup.newColumnType))) {
```

Comment:
We are testing coercion here, so the create type shouldn't same as the insert type